Form/Processor/RoleFormProcessor.php

```diff
@@ -56,12 +56,19 @@ public static function getSubscribedEvents()
 
     public function processDefaultAction(FormActionEvent $event)
     {
-        $this->notify('role.notification.draft_saved', [], 'role');
+        // Don't update anything if we just want to cancel the draft.
+        // Don't do the default action when saving is specified, as it's the same action and shouldn't be done twice.
+        if (in_array($event->getClickedButton(), ['removeDraft', 'saveRole'])) {
```

Comment:
@dpobel Something like this? Ok. Also I can move the explaining comment to the docblock.
```php
if ($event->getClickedButton() === null) {
    $this->processSaveRole($event);
}
```